File: bitrate_halfer.py
"""
reads a PCM audio file from stdin
writes the same file but only half of the bytes (every other byte)
for halfing audio file bitrate
example
 cat file.pcm | python3 bitrate_halfer.py | file_bitrate_halfed.pcm
"""

import sys

audio_bytes = sys.stdin.buffer.read()

modified_data = bytearray()
for i in range(len(audio_bytes) // 2):
    b1 = audio_bytes[2 * i]
    b2 = audio_bytes[2 * i + 1]
    if i % 2 == 0:
        modified_data.append(b1)
        modified_data.append(b2)

sys.stdout.buffer.write(modified_data)

File: bitrate_halfer.lua
--[[
  reads a PCM audio file from stdin
  writes the same file but only half of the bytes (every other byte)
  for halfing audio file bitrate
  example
    cat file.pcm | lua bitrate_halfer.lua | file_bitrate_halfed.pcm
]]

-- 2 bytes per audio file sample
BYTES_PER_SAMPLE = 2

local i = 0
while true do
  local c = io.read(BYTES_PER_SAMPLE)
  if c == nil then break end

  if i % 2 == 0 then
    io.write(c)
  end

  i = i + 1
end
